pkgs/games/spring/default.nix

```diff
@@ -1,63 +1,51 @@
 { lib
 , stdenv
-, fetchFromGitHub
-, cmake
-, xz
-, boost
-, libdevil
-, zlib
-, p7zip
-, openal
-, libvorbis
-, glew
-, freetype
-, xorg
-, SDL2
-, libGLU
-, libGL
 , asciidoc
+, boost
+, cmake
+, curl
 , docbook_xsl
 , docbook_xsl_ns
-, curl
-, makeWrapper
+, fetchurl
+, freetype
+, glew
 , jdk
-, python
-, systemd
+, libdevil
+, libGL
+, libGLU
 , libunwind
-, which
+, libvorbis
+, makeWrapper
 , minizip
+, openal
+, p7zip
+, python3
+, SDL2
+, xorg
+, xz
+, zlib
 , withAI ? true # support for AI Interfaces and Skirmish AIs
 }:
 
 stdenv.mkDerivation rec {
   pname = "spring";
-  version = "105.0.1-${buildId}-g${shortRev}";
-  # usually the latest in https://github.com/spring/spring/commits/maintenance
-  rev = "8581792eac65e07cbed182ccb1e90424ce3bd8fc";
-  shortRev = builtins.substring 0 7 rev;
-  buildId = "1486";
+  version = "106.0";
 
-  # taken from https://github.com/spring/spring/commits/maintenance
-  src = fetchFromGitHub {
-    owner = "spring";
-    repo = pname;
-    inherit rev;
-    sha256 = "05lvd8grqmv7vl8rrx02rhl0qhmm58dyi6s78b64j3fkia4sfj1r";
-    fetchSubmodules = true;
+  src = fetchurl {
+    url = "https://springrts.com/dl/buildbot/default/master/${version}/source/spring_${version}_src.tar.gz";
+    sha256 = "sha256-mSA4ioIv68NMEB72lYnwDb1QOuWr1VHwu4+grAoHlV0=";
```

Comment:
Using the releases avoid the costly `fetchSubmodules` usage and also removes the need of generating the version ourselves.